aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt

```diff
@@ -98,26 +98,26 @@ class AztecMediaSpan(context: Context, drawable: Drawable?, var attributes: Attr
         val sb = StringBuilder()
         sb.append("<")
         sb.append(TAG)
+        sb.append(' ')
 
-        attributes?.let {
-            for (i in 0..attributes!!.length-1) {
-                sb.append(' ')
-                sb.append(attributes!!.getLocalName(i))
-                sb.append("=\"")
-                sb.append(attributes!!.getValue(i))
-                sb.append("\"")
-            }
-        }
+        attributes.removeAttribute("aztec_id")
 
+        sb.append(attributes)
         sb.append("/>")
+
         return sb.toString()
     }
 
     fun onClick(view: View) {
+        // add a unique ID that can identify this image span
```

Comment:
The client has access to the current attributes, so they'd know the value of that ID from the beginning because the library would add it to every element.